parser/src/main/java/log/parser/service/impl/HeroItemsLineParser.java

```diff
@@ -0,0 +1,17 @@
+package log.parser.service.impl;
+
+import java.util.regex.Matcher;
+
+import log.parser.model.HeroItems;
+import log.parser.service.LineParser;
+
+public class HeroItemsLineParser implements LineParser<Object> {
```

Comment:
Must have `@Service` annotation

Must implement

`LineParser<HeroItem>`

There should be a local variable for the pattern:

`private Pattern pattern = Pattern.compile(\\[(\\d{2}):(\\d{2}):(\\d{2}).(\\d{3})\\]\\snpc_dota_hero_(\\S+)\\sbuys\\sitem\\sitem_(\\S+));`

isMatch and parse methods will create a local match variable using:

`Matcher matcher = pattern.matcher(line);`

isMatch will look something like:

```
public boolean isMatch(string line){
    Matcher matcher = pattern.matcher(line);
    return matcher.find();
}
```

And parse method should look like 

```
public HeroItem parse(string line){
    Matcher matcher = pattern.matcher(line);
    HeroItem heroItem = new HeroItem();
    heroItem.setId(UUID.randomUUID());
    heroItem.setItem(matcher.group(0));
    return heroItem;
}
```